### PR TITLE
Test VDS with Helm

### PR DESCRIPTION
**Pull request by benashz**
This PR adds support for running the VDS integration tests against a Helm deployed VSO instance. That means all integration tests should now be tested with Kustomize and Helm.

Additional fixes:
- Move VDS Deployment out of TF
- Test rollout-restart for all VDS test cases

### CODE REVIEW

File: test/integration/modules/vso-helm/main.tf [hunk 1]
```diff
@@ -22,9 +22,19 @@ resource "helm_release" "vault-secrets-operator" {
...
-  set {
-    name  = "defaultAuthMethod.namespace"
-    value = var.operator_namespace
+  dynamic "set" {
+    for_each = var.k8s_auth_default_mount != "" ? [""] : []
+    content {
+      name  = "defaultAuthMethod.mount"
+      value = var.k8s_auth_default_mount
+    }
+  }
```
Comment (tvoran) [inline test/integration/modules/vso-helm/main.tf:31]: Just making sure I understand, this sets `k8s_auth_default_mount` in the chart values if it's not blank?

File: test/integration/vaultdynamicsecret_integration_test.go [hunk 2]
```diff
@@ -50,12 +45,9 @@ func TestVaultDynamicSecret(t *testing.T) {
...
 
-	tfDir, err := files.CopyTerraformFolderToDest(
-		path.Join(testRoot, "vaultdynamicsecret/terraform"),
-		tempDir,
-		"terraform",
-	)
-	require.Nil(t, err)
+	tfDir := copyTerraformDir(t, path.Join(testRoot, "vaultdynamicsecret/terraform"), tempDir)
+	copyModulesDir(t, tfDir)
+	chartDestDir := copyChartDir(t, tfDir)
```
Comment (tvoran) [inline test/integration/vaultdynamicsecret_integration_test.go:50]: I'm curious why this is working off a copy of the chart dir, does the test modify the chart in some way?

**Review by tvoran — APPROVED** [review]

File: test/integration/modules/vso-helm/main.tf (hunk 1 above)
Comment (benashz) [reply test/integration/modules/vso-helm/main.tf:31]: yes

File: test/integration/vaultdynamicsecret_integration_test.go (hunk 2 above)
Comment (benashz) [reply test/integration/vaultdynamicsecret_integration_test.go:50]: By working from a copy, it allows you to be able to modiy the chart while a test is running without affecting that test run. The test should never modify the contents of the chart.

File: test/integration/modules/vso-helm/main.tf (hunk 1 above)
Comment (benashz) [reply test/integration/modules/vso-helm/main.tf]: Updated with ef59401fddfa1f0cf01bb22508792a8e3840d6b8

File: test/integration/vaultdynamicsecret/terraform/locals.tf [hunk 3]
```diff
@@ -20,4 +20,10 @@ locals {
   db_role             = "dev-postgres"
   db_role_static      = "${local.db_role}-static"
   db_role_static_user = "${local.db_role_static}-user"
+
+  # transit locals
+  operator_service_account_name = "${local.name_prefix}-operator"
+
+  #
```
[the remark this answers is not in the document]
Comment (benashz) [reply test/integration/vaultdynamicsecret/terraform/locals.tf]: Fixed with ef59401fddfa1f0cf01bb22508792a8e3840d6b8

**Pull request merged**